app/validators/user_validator.rb

```diff
@@ -36,4 +37,16 @@ def valid_phone_number_if_receive_sms_notifications(record)
       record.errors.add(:base, " Must add a valid phone number to receive SMS notifications.")
     end
   end
+
+  def valid_date_of_birth(date_of_birth, record)
+    return true unless date_of_birth.present?
+
+    # Date of birth must be in the past
+    if date_of_birth.to_time >= Time.zone.now
+      record.errors.add(:base, " Date of birth must be in the past.")
+      false
+    else
+      true
+    end
+  end
```

Comment:
Again, I'm new here so take with a giant grain of salt.

I don't believe custom validators need to return true or false. They only need to add an error to the record or not. with that said I here's a suggested refactoring.

```ruby
  def valid_date_of_birth(date_of_birth, record)
    return unless date_of_birth.present?

    record.errors.add(:base, "Date of birth must be in the past.") if date_of_birth.past?
                                                     #                              ^^^^^
                                                     # I believe this can do the check for in the past.
  end
```

### References
validators ref: https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations
`past?` ref: https://api.rubyonrails.org/v7.0/classes/DateAndTime/Calculations.html#method-i-past-3F

Reply:
Good catch, thanks